process-engine/consumer_api_http
# ♻️ Refactor socket io handling

**Changes:**

1. Use the IdentityService to create Identities. This avoids the need to manually decode a requesting users token, in order to get the associated userId.
2. Implement Socket.IO endpoint disposal:
    - When a client disconnects, the subscriptions created specifically for that client will get removed from the event aggregator.
    - When the endpoint itself is disposed, all socket.IO clients get disconnected and all subscriptions created by the endpoint will be removed.
3. Remove the redundant `IConnection` type.
4. Create an internal Subscription storage that keeps track of all Subscriptions. The storage is split into two scopes:
    - Endpoint-Scope: Stores all global subscriptions that are used commonly for all clients.
    - Client-Scope: Stores all subscriptions for the individual clients. 

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/#135
Part of https://github.com/process-engine/process_engine_runtime/issues/#157
Part of https://github.com/process-engine/process_engine_runtime/issues/#206

PR: #26

## How can others test the changes?

- Connect a client.
- Trigger some notifications for that client.
- Disconnect the client.
- Trigger the same notifications again.
- Notice that the corresponding subscriptions are no longer active.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
